bing_rewards/__init__.py

```diff
@@ -190,55 +190,69 @@ def main():
     Setup listener callback for ESC key.
     """
     options = app_options.get_options()
-
     words_gen = word_generator()
 
-    def desktop():
+    def desktop(profile=''):
         # Complete search with desktop settings
         count = options.count if 'count' in options else options.desktop_count
-        print(f'Doing {count} desktop searches')
+        profile_msg = f" using profile '{profile}'" if profile else ''
+        print(f'Doing {count} desktop searches{profile_msg}')
 
-        search(count, words_gen, options.desktop_agent, options)
+        temp_options = options
+        temp_options.profile = profile
+        search(count, words_gen, options.desktop_agent, temp_options)
         print('Desktop Search complete!\n')
 
-    def mobile():
+    def mobile(profile=''):
         # Complete search with mobile settings
         count = options.count if 'count' in options else options.mobile_count
-        print(f'Doing {count} mobile searches')
+        profile_msg = f" using profile '{profile}'" if profile else ''
+        print(f'Doing {count} mobile searches{profile_msg}')
 
-        search(count, words_gen, options.mobile_agent, options)
+        temp_options = options
+        temp_options.profile = profile
+        search(count, words_gen, options.mobile_agent, temp_options)
         print('Mobile Search complete!\n')
 
-    def both():
-        desktop()
-        mobile()
+    def both(profile=''):
+        desktop(profile)
+        mobile(profile)
 
     # Execute main method in a separate thread
     if options.desktop:
-        target = desktop
+        target_func = desktop
     elif options.mobile:
-        target = mobile
+        target_func = mobile
     else:
         # If neither mode is specified, complete both modes
-        target = both
-
-    # Start the searching in separate thread
-    search_thread = threading.Thread(target=target, daemon=True)
-    search_thread.start()
-
-    print('Press ESC to quit searching')
-
-    try:
-        # Listen for keyboard events and exit if ESC pressed
-        while search_thread.is_alive():
-            with keyboard.Events() as events:
-                event = events.get(timeout=0.5)  # block for 0.5 seconds
-                # Exit if ESC key pressed
-                if event and event.key == Key.esc:
-                    print('ESC pressed, terminating')
-                    break
-    except KeyboardInterrupt:
-        print('CTRL-C pressed, terminating')
+        target_func = both
+
+    # If no profiles specified, run once with default (empty) profile
+    profiles = options.profile if options.profile else ['']
+
+    for profile in profiles:
+        # Start the searching in separate thread
+        search_thread = threading.Thread(target=lambda p=profile: target_func(p), daemon=True)
```

Comment:
[`Thread` can take the `args=` argument](https://docs.python.org/3/library/threading.html#threading.Thread) to pass to the target function,
so this becomes simpler like: 
```python
 search_thread = threading.Thread(target=target_func, args=(profile,), daemon=True)```